src/tablestore.rs

```diff
@@ -213,17 +205,25 @@ impl TableStore {
                 _ => SlateDBError::ObjectStoreError(e),
             })?;
 
-        self.cache_filter(*id, encoded_sst.filter);
+        self.cache_filter(
+            *id,
+            encoded_sst.info.borrow().filter_offset(),
```

Comment:
It's a bit weird that we're using the filter/index offset as the id for filters/indexes, but we use the block num for the data blocks. I think it works, because given our current SST format that places indexes and filters at the end, the index/filter offset will be strictly greater than the largest block num. This is leaking the format into the id though. If we were to use a format that put the filter/index in the beginning this wouldn't work since the offset could collide with a block id. tl;dr - can we either use the offset for everything, use some special block num to indicate a filter or index, or include the block type in the key?

Reply:
I think using offset for everything is cleaner unless you have a preference for one of the other options. Will update accordingly.

Reply:
+1 for offset-for-everything. That feels the most straightforward to me. Do you mind updating the `CachedBlock .Block` cache key usage accordingly?

Reply:
I did, should be reflected now

Reply:
Ah, I see you already made the update! Nice. :D